modules/mono/build_scripts/build_assemblies.py

```diff
@@ -338,7 +339,7 @@ def generate_sdk_package_versions():
     )
     os.makedirs(generators_dir, exist_ok=True)
 
-    with open(os.path.join(generators_dir, "Common.Constants.cs"), "w") as f:
+    with open(os.path.join(generators_dir, "Common.Constants.cs"), "w", newline="\n") as f:
```

Comment:
Nice catch. Looking at other python scripts in the codebase, it looks like none of them are specifying the newline character. Also, they tend to specify the encoding (`utf-8`) which we may also want to do here.

Reply:
I'm adding the encoding to both `open()` call we make in this file, I think we care about the newline only when we write sources, to not flood output from `dotnet format` when running on Windows.

Reply:
From what I see, the other usages of `open()` in the codebase seem to be adding `\n` manually so I think we're good.